components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -446,6 +454,18 @@ impl FileLoader {
         })
     }
 
+    pub fn read<T: serde::de::DeserializeOwned>(&self, file: &FilePath) -> Result<T> {
+        let string = self
+            .read_to_string(file)
+            .map_err(|e| FMLError::InvalidPath(format!("{file}: {e}")))?;
+
+        Ok(match file.extension().map(str::to_lowercase).as_deref() {
+            Some("json") => serde_json::from_str(&string)?,
+            Some("yaml") | Some("yml") => serde_yaml::from_str(&string)?,
```

Comment:
Hi! I'm sorry for the drive-by comment, but I saw this go by, and it reminded me of something I read forever ago about how YAML was designed to be a superset of JSON—so a YAML parser could also parse JSON out of the box.

I double-checked the [YAML spec](https://yaml.org/spec/1.2.2/) to see if I was completely off, and found:

> The YAML 1.2 specification was published in 2009. Its primary focus was making YAML a strict superset of JSON.

...So I was wondering if it makes sense to use `serde_yaml::from_str` unconditionally, instead of branching on the extension and deciding whether to use `serde_json` or `serde_yaml`.

Reply:
> Hi! I'm sorry for the drive-by comment, but I saw this go by, 

@linabutler I'm happy for comments and drive bys from anyone, but especially welcome them from you. :D

> and it reminded me of something I read forever ago about how YAML was designed to be a superset of JSON—so a compliant YAML parser could also parse JSON out of the box.

You're absolutely right! I've deleted the `match` in this method. I've kept the rest of the PR the same:

- the new `read` method clears up a lot of duplicated parsing
- the `extension` method is well tested and may prove useful in future cleanups.